write-fonts/src/pens.rs

```diff
@@ -169,31 +169,40 @@ pub struct ReverseContourPen<'a, T: Pen> {
 ///
 ///  FontTools version in
 /// <https://github.com/fonttools/fonttools/blob/78e10d8b42095b709cd4125e592d914d3ed1558e/Lib/fontTools/pens/reverseContourPen.py#L25>
+///
+///  We differ from FontTools in that we start from the last point of the original contour, not the first.
+///  This way the implied closepath line-to of the original contour will become the implied closepath of
+///  the reversed contour. Credits to Behdad Esfahbod for the idea, see
+///  <https://github.com/fonttools/fonttools/issues/3093#issuecomment-1528134312>
 fn flush_subpath<T: Pen>(commands: &[PenCommand], pen: &mut T) -> Result<(), ContourReversalError> {
     if commands.is_empty() {
         return Ok(());
     }
 
-    let mut commands = commands;
-    let mut reversed = Vec::new();
-
     // subpath must start with a move, and by definition it can't have any other move
     let PenCommand::MoveTo { x, y } = commands[0] else {
         return Err(ContourReversalError::SubpathDoesNotStartWithMoveTo);
     };
     let (start_x, start_y) = (x, y);
 
-    // When reversed, the move is to the end point of the last command
-    // in a typical [move, ..., close] structure, end point == start point
-    let (end_x, end_y) = commands
-        .last()
-        .unwrap()
-        .end_point()
-        .unwrap_or((start_x, start_y));
+    // When reversed, the move is to the end point of the last segment,
+    // i.e. the last command in an open contour, or the penultimate command
+    // in a closed contour.
+    let is_closed = *commands.last().unwrap() == PenCommand::Close;
+    let last_segment_idx = if is_closed {
```

Comment:
I think it would be worth a comment to note that -2 must be a valid index because we know that first is move and last is close so there must be at least move/close. Since you are headed on vacation I will add the comment and merge.